core/lib/dal/src/proof_generation_dal.rs

```diff
@@ -183,4 +183,34 @@ impl ProofGenerationDal<'_, '_> {
 
         result
     }
+
+    /// Checks if proof is ready and return proof's blob url.
+    /// There are 3 possible outcomes:
+    ///     None -> means that proof is not ready.
+    ///     Some(None) -> means that proof was skipped
+    ///     Some("path") -> means that proof was processed and is available at path.
```

Comment:
I would probably create a data type for this, but feel free to leave as is - the return type is clear from the comment

Reply:
> I would probably create a data type for this

+1